src/amo/components/AddonBadges/index.js

```diff
@@ -149,13 +143,14 @@ export class AddonBadgesBase extends React.Component<InternalProps> {
 
     const averageDailyUsers = addon.average_daily_users;
 
-    const userCount = i18n.formatNumber(averageDailyUsers);
-    const userTitle =
+    const userLabel =
       averageDailyUsers > 0
-        ? i18n.ngettext('User', 'Users', averageDailyUsers)
+        ? [
+            i18n.formatNumber(averageDailyUsers),
+            i18n.ngettext('User', 'Users', averageDailyUsers),
+          ].join(' ')
```

Comment:
In general, you should not concatenate or join localized strings. The way to translate this string is this example: https://github.com/mozilla/addons-frontend/blob/17b5f91f196619cdf455c4ac19e3ae75b641a5c6/src/amo/components/UserCollection/index.js#L45-L48

Reply:
Also, since the existing test coverage didn't catch this, we should improve the test in this PR.

Reply:
Ah great point. Yeah this is supposed to be a WIP Pr. I have not even touched tests.